### PR TITLE
Query update animals table

### PR DESCRIPTION
**Pull request by SimonGideon**
# Project requirements
Add a column species of type string to your animals table. Modify your schema.sql file.
Insert the following data:
Animal: His name is Charmander. He was born on Feb 8th, 2020, and currently weighs -11kg. He is not neutered and he has never tried to escape.
Animal: Her name is Plantmon. She was born on Nov 15th, 2021, and currently weighs -5.7kg. She is neutered and she has tried to escape 2 times.
Animal: His name is Squirtle. He was born on Apr 2nd, 1993, and currently weighs -12.13kg. He was not neutered and he has tried to escape 3 times.
Animal: His name is Angemon. He was born on Jun 12th, 2005, and currently weighs -45kg. He is neutered and he has tried to escape once.
Animal: His name is Boarmon. He was born on Jun 7th, 2005, and currently weighs 20.4kg. He is neutered and he has tried to escape 7 times.
Animal: Her name is Blossom. She was born on Oct 13th, 1998, and currently weighs 17kg. She is neutered and she has tried to escape 3 times.
Animal: His name is Ditto. He was born on May 14th, 2022, and currently weighs 22kg. He is neutered and he has tried to escape 4 times.
Remember to add these insertions to your data.sql file.
Inside a transaction update the animals table by setting the species column to unspecified. Verify that change was made. Then roll back the change and verify that the species columns went back to the state before the transaction.
Inside a transaction:
Update the animals table by setting the species column to digimon for all animals that have a name ending in mon.
Update the animals table by setting the species column to pokemon for all animals that don't have species already set.
Commit the transaction.
Verify that change was made and persists after commit.
Take a screenshot of the results of your actions.
Now, take a deep breath and... Inside a transaction delete all records in the animals table, then roll back the transaction.
After the rollback verify if all records in the animals table still exists. After that, you can start breathing as usual ;)
Take a screenshot of the results of your actions.
Inside a transaction:
Delete all animals born after Jan 1st, 2022.
Create a savepoint for the transaction.
Update all animals' weight to be their weight multiplied by -1.
Rollback to the savepoint
Update all animals' weights that are negative to be their weight multiplied by -1.
Commit transaction
Take a screenshot of the results of your actions.
Remember to add all these transactions to your queries.sql file.
Write queries to answer the following questions:
How many animals are there?
How many animals have never tried to escape?
What is the average weight of animals?
Who escapes the most, neutered or not neutered animals?
What is the minimum and maximum weight of each type of animal?
What is the average number of escape attempts per animal type of those born between 1990 and 2000?
Take a screenshot of the results of your queries.
Remember to add these queries to your queries.sql file.
Remember to add screenshots with the results of your queries and with the results of your play with transactions to your Pull Request description

## Trasaction
Delete
roll back
species

## Queries
average
count
escape attampts
image
image
image

### CODE REVIEW

File: queries.sql [hunk 1]
```diff
@@ -9,3 +9,38 @@ SELECT * FROM animals WHERE neutered=true;
...
+UPDATE animals SET weight_kg = weight_kg * -1;
+ROLLBACK TO my_savepoint;
+UPDATE animals SET weight_kg = weight_kg * -1 WHERE weight_kg < 0;
+COMMIT;
+
+-- Queries
+SELECT COUNT(name) FROM animals;
+SELECT COUNT(name) FROM animals WHERE escape_attempts = 0;
+SELECT AVG(weight_kg) FROM animals;
+SELECT neutered, COUNT(*) FROM animals GROUP BY neutered ORDER BY COUNT(*) DESC LIMIT 1;
```
Comment (Gambit142) [inline queries.sql:41]: - [ ] This query will return the wrong result. The query should return which group escapes the most in terms of _**neutered**_ animals and _**non-neutered**_ animals. To return the right data, you would have to select the `neutered` column and the `AVG(escape_attempts)` or `MAX(escape_attempt)` and then group by neutered. All you have to do is replace `COUNT(*)` with either `AVG(escape_attempts)` or `MAX(escape_attempt)`

**Review by Gambit142 — CHANGES_REQUESTED** [review]
Hi @SimonGideon 👋, 

Good job so far!

## Highlights
- README.md file is professional ✔️ 
- Everything that creates or changes the table structure is saved in the schema.sql file.
- Everything that populates tables with data is saved in the data.sql file.
- Everything that is a query that helps to analyze is saved in the queries.sql file.

There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Kindly check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

You can also consider:

N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### Please, remember to tag me in your question so I can receive the notification. You can tag me by typing @Gambit142

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: data.sql [hunk 2]
```diff
@@ -6,3 +6,8 @@ VALUES('Agumon', DATE '2020-02-03',0, false, 10.23);
 VALUES('Gabumon', DATE '2018-11-15',2,true,8);
 VALUES('Pikachu', DATE '2021-01-07',1,false,15.04);
 VALUES('Devimon', DATE '2017-05-12',5,true,11);
+
+INSERT INTO animals (
+name, date_of_birth, escape_attempts, neutered, weight_kg)
+VALUES('Charmander', DATE '2020-02-08',0,false,11),('Plantmon',DATE '2021-11-15',2, TRUE, 5.7),('Squirtle', DATE '1993-02-04',3,FALSE,12.13),('Angemon',DATE '2003-06-12',1,TRUE,45),('Boarmon',DATE '2005-06-07',7,TRUE,20.4),('Blossom',DATE '1998-10-13',3,TRUE,17),('Ditto', DATE '2022-05-14',4,TRUE,22);
+
```
[the remark this answers is not in the document]
Comment (SimonGideon) [reply data.sql:13]: image

image

:+1: @Gambit142 I request you check that the values of `weight_kg` of `Blossom` and `Boarmon` are correct as Outlined in the Question.

File: data.sql [hunk 3]
```diff
@@ -6,3 +6,8 @@ VALUES('Agumon', DATE '2020-02-03',0, false, 10.23);
 VALUES('Gabumon', DATE '2018-11-15',2,true,8);
 VALUES('Pikachu', DATE '2021-01-07',1,false,15.04);
 VALUES('Devimon', DATE '2017-05-12',5,true,11);
```
[the remark this answers is not in the document]
Comment (SimonGideon) [reply data.sql:8]: :smile: Noted. Corrected in the future commits.

File: queries.sql (hunk 1 above)
Comment (SimonGideon) [reply queries.sql:41]: Done

**Review by Davidosky007 — APPROVED** [review]
Hi @SimonGideon 👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
* Well written PR description 👍 
* Good looking readme file 👍 
* Proper use of Gitflow 👍 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: data.sql (hunk 2 above)
[the remark this answers is not in the document]
Comment (Gambit142) [reply data.sql:13]: Hi @SimonGideon 👋, please read my statement again. I said `apart from`. You ought to input negative values for most of the animals in this milestone. 👍

[the remark this answers is not in the document]
Comment (SimonGideon) [reply data.sql:13]: @Gambit142 :slightly_smiling_face: Got it man. Thanks

**Pull request merged**